front/src/components/student-registration/student-registration-form/student-registration-form.js

```diff
@@ -0,0 +1,203 @@
+import React, { Component } from 'react';
+import obj from '../../../common/redux/actions';
+import { Row, Form, Typography } from 'antd';
+import { Col, Input, Button } from '../student-registration-form/style';
+
+const { Title } = Typography;
+
+const layout = {
+  labelCol: { span: 8 },
+  wrapperCol: { span: 16 },
+};
+
+const tailLayout = {
+  wrapperCol: {
+    offset: 9,
+    span: 16,
+  },
+};
+
+class StudentRegistrationPage extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.validateEng = this.validateEng.bind(this);
+    this.validateUkr = this.validateUkr.bind(this);
+    this.validatePhoneNumber = this.validatePhoneNumber.bind(this);
+    this.confirmPassword = this.confirmPassword.bind(this);
+  }
+
+  handleSubmit(data) {
+    data.groupToken = this.props.token;
+
+    this.props.dispatch(obj.studentRegistered(data));
+  }
+
+  async validateEng(_, value) {
+    const engRegexp = /^[A-Z]{1}([a-z]{1,})?$/;
+    if (engRegexp.test(value)) {
+      return Promise.resolve();
+    } else {
+      return Promise.reject('Not valid name or surname in English');
+    }
+  }
+
+  async validateUkr(_, value) {
+    const ukrRegexp = /^[А-ЯІЇЄ]{1}([а-яіїє]{1,})?$/;
+    const excludeRuLetters = /^[^ЫЭЪЁ]{1}([^ыэъё]{1,})?$/;
+
+    if (ukrRegexp.test(value) && excludeRuLetters.test(value)) {
+      return Promise.resolve();
+    } else {
+      return Promise.reject('Not valid name or surname in Ukrainian');
+    }
+  }
+
+  async validatePhoneNumber(_, value) {
+    const phoneNumberRegexp = /\+380\d{9}/;
+
+    if (phoneNumberRegexp.test(value)) {
+      return Promise.resolve();
+    } else {
+      return Promise.reject('Not valid phone number');
+    }
+  }
+
+  async confirmPassword({ getFieldValue }) {
+    return {
+      validator(rule, value) {
+        if (!value || getFieldValue('password') === value) {
+          return Promise.resolve();
+        }
+        return Promise.reject('The two passwords that you entered do not match!');
+      },
+    };
+  }
```

Comment:
Maybe validation functions should be outside of this component as validation-helper?

Reply:
Maybe you are right, because in the future we will have more components that could use same validation functions. I am starting to work on new mentors component, that has simular logic. I will try to refactor it at that PR